### ISSUE TITLE
`fileSystems.<name>.label` not compatible with PR #172

### ISSUE THREAD

**Issue by herrwiese** [nix-community/impermanence#218]
When specifying a file system by label using `fileSystems."/foo".label = "foo";` instead of using the `device` attribute, this now fails to evaluate as PR #172 expects the latter to be set.  This is similar to the ZFS problem described in #216.

**Comment by talyz** [comment]
Please provide more detailed info - at least an example configuration and the evaluation error.

**Comment by MatthewCash** [comment]
Essentially, #172 causes an evaluation error whenever a filesystem's device attr is unset or null even though this is a valid configuration (see [fileSystems.<name>.device](https://search.nixos.org/options?channel=unstable&show=fileSystems.%3Cname%3E.device&from=0&size=50&sort=relevance&type=packages&query=filesystems.%3Cname%3E.device)).

Not including `device` is common when mounting partitions based on FS or partition labels.

A small example:
```nix
fileSystems.nix = {
  label = "nix";
  fsType = "btrfs";
  mountPoint = "/nix";
  options = [ "rw" "noatime" ];
};
```

and then trying to evaluate 
```nix
nixosConfigurations.systemname.config.boot.initrd.systemd.services.create-needed-for-boot-dirs
```

results in 
```nix
{ after = error:
       … while evaluating the attribute 'value'

         at /nix/store/fpivx4sjcp2vk4rp9nhliln5cwcp3kc6-source/lib/modules.nix:821:9:

          820|     in warnDeprecation opt //
          821|       { value = addErrorContext "while evaluating the option `${showOption loc}':" value;
             |         ^
          822|         inherit (res.defsFinal') highestPrio;

       … while calling the 'addErrorContext' builtin

         at /nix/store/fpivx4sjcp2vk4rp9nhliln5cwcp3kc6-source/lib/modules.nix:821:17:

          820|     in warnDeprecation opt //
          821|       { value = addErrorContext "while evaluating the option `${showOption loc}':" value;
             |                 ^
          822|         inherit (res.defsFinal') highestPrio;

       (stack trace truncated; use '--show-trace' to show the full trace)

       error: cannot coerce null to a string
```

**Comment by talyz** [comment]
Right, I noticed now that handling for that was only done in the script and not in the dependency list. This should be fixed by #221, so please try it out.
